base/dict.jl

```diff
@@ -237,7 +232,7 @@ end
     h.count = count
     h.ndel = 0
     h.maxprobe = maxprobe
-    @assert h.age == age0
+    @assert h.age == age0 "Muliple concurent writes to Dict detected!"
```

Comment:
move this above the assignments, so we only do the mutation if things are not broken?

Reply:
It's only a sanity check for now and thus can be changed anywhere. I've also added one aging for more safety.